.circleci/config.yml

```diff
@@ -76,6 +78,22 @@ jobs:
           paths:
             - ~/.cache/yarn
             - node_modules
+
+  schematics-core-copy:
+    <<: *run_in_node_latest_browsers
+    steps:
+      - checkout
+      - restore_cache:
+          key: *cache_key
+      - run: yarn
+      - run: yarn copy:schematics
+      - run: git diff --name-only --exit-code ./modules
+      - save_cache:
```

Comment:
We don't have to save the cache I believe.

Reply:
Actually, after giving it more thought:

The test job is running in parallel with your new build step and these rely both on the yarn cache.
Personally it makes more sense to me to first check if the files are copied and if these are,l to run the tests. Otherwise the tests may result in a false positive - a test may turn green until you copied the schematic files. 

To fix this we can add the install step as you mentioned, then run the check, then run the tests.
Or we could add the git diff check in the current test step.

Perhaps we could add this in the BAZEL flow, but I don't know anything about it 😄

Reply:
@timdeschryver 
I consider two ways to resolve the matter you mentioned.
The first way is configuring `check-schematics-core-copied` as dependency of `test`.
We check if the files are copied and after that run tests.

```
workflows:
  version: 2
  build-test-deploy:
    jobs:
      - lint
      - bazel
      - check-schematics-core-copied
      - test
          requires:
            - check-schematics-core-copied
      - deploy:
          requires:
            - test
          filters:
            branches:
              only: master
```

In this case, we need move `save_cache` to `check-schematics-core-copied` from 'test' because `check-schematics-core-copied` is executed certainly before `test`.

The second way is create new job to install and save npm dependencies as I mentioned before.

```
yarn-install:
  <<: *run_in_node_latest_browsers
  steps:
    - checkout
    - restore_cache:
        key: *cache_key
    - run: yarn
    - save_cache:
        key: *cache_key
        paths:
        - ~/.cache/yarn
        - node_modules

workflows:
  version: 2
  build-test-deploy:
    jobs:
      - lint
      - bazel
      - check-schematics-core-copied
          requires:
            - yarn-install
      - test
          requires:
            - yarn-install
      - deploy:
          requires:
            - check-schematics-core-copied
            - test
          filters:
            branches:
              only: master
```

I think it's preffer to use this way because it can apply to other jobs created in the future.
Please tell me your opinion :grinning:

Reply:
Yep I agree, and I also like option 2 more.

You could do the following to prevent false positives:

```
- test
  requires:
    - check-schematics-core-copied
```

And to be clear, what I meant with the second option I mentioned in my comment above is to modify the test step to the following:
```
test:
    <<: *run_in_node_latest_browsers
    steps:
      - checkout
      - restore_cache:
          key: *cache_key
      - run: yarn
      - run: yarn run ci
      - run: yarn copy:schematics
      - run: git diff --name-only --exit-code ./modules
      - run: yarn run example:build:prod
      - run: yarn run example:test --watch=false
      - save_cache:
          key: *cache_key
          paths:
            - ~/.cache/yarn
            - node_modules
````

I think this is the preferred way because the current steps aren't split either.

Reply:
Indeed, it's more simpler and understandable for me :smile:.
Please see my update since I push additionally.

In the example code, `run: yarn run ci` executed before `yarn copy:schematics` but I think it should run after `yarn copy:schematics` because it executes test command internally.
Is this correct?

Reply:
@timdeschryver 
Could you please reply to me? :man-bowing:

Reply:
😅 sorry @kouMatsumoto